catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1467,7 +1467,7 @@ public QueryResponse query(QueryRequest queryRequest, FederationStrategy strateg
         } catch (RuntimeException re) {
             LOGGER.warn("Exception during runtime while performing query", re);
             throw new UnsupportedQueryException("Exception during runtime while performing query");
-
+            //TODO -- Why transform all runtime exceptions to unsupported query exceptions?
```

Comment:
Things like this are done so that the calling code can expect certain types of exceptions back instead of receiving an entire range of exceptions. It also reduces the likelihood of calling code exposing that internal runtime exception to the caller.